src/models/resourcekindhandler.ts

```diff
@@ -19,10 +19,14 @@ interface RefMapper {
         sourceResource: K8sResource,
         targetResource: K8sResource,
         value: string,
-        siblingValues: Record<string, string>
+        siblingValues: Record<string, string>,
+        properties?: any
       ) => boolean
     >;
 
+    // optional matcher configuration properties that will be passed to each correspondingly named matcher
+    matcherProperties?: Record<string, any>;
```

Comment:
Maybe we should refactor this to:
```ts
type Matcher = 'kindMatcher' | 'groupMatcher' // ... add additional matcher types

import {JsonObject} from "type-fest"; // using this type to ensure RefMappers will remain serializable

interface RefMapper {
  // ...
  matcherProperties: Record<Matcher, JsonObject>
  // ...
}

Reply:
sure - I think this would be fine until we allow plugins to provide their own matchers... !?

Reply:
actually - the key of this map is dynamic - i.e. depends on the property to be matched - so we can't create a type with a fixed set of names. I'll add the JsonObject though 👍